tests/ssl-opt.sh

```diff
@@ -1696,6 +1697,7 @@ run_test    "Connection ID, 3D+MTU: Cli+Srv enabled, Cli+Srv CID nonempty" \
             -s "ignoring unexpected CID"
 
 requires_config_enabled MBEDTLS_SSL_DTLS_CONNECTION_ID
+requires_config_disabled MBEDTLS_SSL_CONF_CID_LEN
```

Comment:
Note: I think ideally this could be replaced with a requirement that either `MBEDTLS_SSL_CONF_CID_LEN` is undefined, or it is defined and equal to 4. However I don't think the current `requires_config_xxx` macro allow that easily, and I think it's acceptable for this PR to skip a bit more often that strictly necessary, so I'm not requesting any change here, just noting. We might want to improve the condition in a later PR, though.

Reply:
I agree, there should be a flexible way to skip the tests, or alter the parameters depending on the configuration.